src/components/code-editor/index.js

```diff
@@ -4,43 +4,54 @@ module.exports = {
         return {
             showSpinner: false,
 	        expectingSave: false,
-	        saving: false
+	        saving: false,
+            currentFile: null,
+            currentParent: null
         }
     },
-    props: ['context', 'file', 'parent'],
+    props: ['context', 'file', 'parent', 'messages'],
     created: function() {
-        this.startListener();
+        let that = this;
+        that.currentParent = this.parent;
+        that.currentFile = this.file;
+        this.currentParent.getLatest(this.context.network).thenApply(function(updatedParent) {
```

Comment:
This shouldn't be done via the parent. That's what I meant about cleaning up the FileWrapper code. We should be able to modify a file from the file itself. This make it easier to reason about things for example when the file is a different writer to its parent